.github/workflows/test.yml

```diff
@@ -390,3 +390,41 @@ jobs:
       uses: Ilshidur/action-discord@0.3.2
       with:
         args: "Master tests failed: <https://github.com/${{github.repository}}/actions/runs/${{github.run_id}}>"
+  test-build-docker-gnu:
+    name: build-docker
+    runs-on: ubuntu-latest
+    steps:
+    - name: install qemu
+      uses: docker/setup-qemu-action@v1
+    - name: install buildx
+      uses: docker/setup-buildx-action@v1
+      id: buildx
+      with:
+        install: true
+    - name: checkout code
+      uses: actions/checkout@v2
+    - name: build for amd64
+      run: |
+        docker buildx build \
+          --platform linux/amd64 \
+          --build-arg RUST_VERSION=$(cat rust-toolchain.toml | grep channel | cut -d '"' -f 2) \
+          -t jdrouet/vector:gnu \
+          -f scripts/docker/gnu.Dockerfile \
+          .
+    - name: build for arm64
+      run: |
+        docker buildx build \
+          --platform linux/arm64 \
+          --build-arg RUST_VERSION=$(cat rust-toolchain.toml | grep channel | cut -d '"' -f 2) \
+          -t jdrouet/vector:gnu \
+          -f scripts/docker/gnu.Dockerfile \
+          .
+    - name: build for arm32v7
+      run: |
+        docker buildx build \
+          --platform linux/arm/v7 \
+          --build-arg RUST_VERSION=$(cat rust-toolchain.toml | grep channel | cut -d '"' -f 2) \
+          -t jdrouet/vector:gnu \
+          -f scripts/docker/gnu.Dockerfile \
+          .
```

Comment:
Presumably once this is working properly we can turn this into a matrix?

Reply:
Actually, it's possible to do `--platforms linux/amd64,linux/arm64,linux/arm/v7` but it runs in parallel and increases the total build time.
It's possible, with buildx, to use a dedicated engine for a specific platform: have a `multiarch` builder that forwards the `arm64` build to an engine that runs on a `arm64` machine, etc.

Reply:
I was just thinking we could reduce the duplication with the `matrix` CI feature, rather than duplicating the yaml/script